examples/02_decoding/plot_haxby_grid_search.py

```diff
@@ -74,6 +74,14 @@
 # on nested cross-validation.
 from nilearn.decoding import Decoder
 
+# We provide a grid of hyperparameter values to the Decoder's internal
+# cross-validation. If no param_grid is provided, the Decoder will use a
+# default grid with sensible values for the chosen estimator
+param_grid = {
+    "C": [1, 10, 100, 1000],
+    "intercept_scaling": [1, 10],
```

Comment:
@jeromedockes I agree that it is a more interesting hyperparameter, but "penalty" is supposed to be fixed to L2 for the "svc" estimator (see [here](https://github.com/nilearn/nilearn/blob/983accf6/nilearn/decoding/decoder.py#L54)). I think for L1 penalty the user is expected to use the "svc_l1" estimator instead

Reply:
just because it is initialized with that value doesn't mean it's fixed, `set_params` will change it -- note `penalty="l2"` is the default so it is repeated in the line you linked for clarity but if it wasn't there it wouldn't make a difference. otherwise if not the penalty maybe we can add the loss to the grid?

Reply:
Ok, I changed it to this:
```python
param_grid = [
    {
        "penalty": ["l2"],
        "dual": [True],
        "C": [100, 1000],
    },
    {
        "penalty": ["l1"],
        "dual": [False],
        "C": [100, 1000],
    },
]
```

Note that if `param_grid` is a list of dicts and the dicts have different keys, the fitted Decoder's `cv_params_` output can contain lists that have less elements than the number of CV folds. For example, if `param_grid` is
```python
param_grid = [
    {
        "penalty": ["l2"],
        "C": [100, 1000],
    },
    {
        "penalty": ["l1"],
        "dual": [False],    # "dual" is not in the first dict
        "C": [100, 1000],
    },
]
```

Then it is possible that `decoder.cv_params_["shoe"]` has something like
```python
{
    'C': [1000, 1000, 100, 100, 1000], 
    'dual': [False, False, False],    # less than 5, not aligned with the other hyperparameter values
    'penalty': ['l1', 'l2', 'l1', 'l1', 'l2'],
}
```

I think this should be considered a bug

Reply:
I think so too!